File: src/types/budget.ts
export type BudgetMember = {
  id: string;
  email: string | null;
  name: string | null;
};

export type BudgetIncome = {
  id: string;
  source: string;
  amount: number;
};

export type BudgetFixedExpense = {
  id: string;
  name: string;
  amount: number;
  enabled: boolean;
};

export type BudgetLedgerEntry = {
  id: string;
  amount: number;
  category: string;
  merchant?: string | null;
  date: string;
  isOneTime?: boolean;
  tags?: string[];
};

export type BudgetGoal = {
  id: string;
  name: string;
  targetAmount: number;
  currentBalance: number;
  monthlyContribution: number;
  createdAt: string;
  updatedAt: string;
};

export type BudgetCustomCategory = {
  id: string;
  value: string;
  label: string;
  emoji?: string | null;
  memberId: string | null;
  createdAt: string;
};

export type BudgetCategoryRule = {
  id: string;
  memberId: string | null;
  operator: "contains" | "starts_with" | "equals";
  pattern: string;
  categoryValue: string;
  createdAt: string;
};

export type BudgetCustomTag = {
  id: string;
  value: string;
  memberId: string | null;
  createdAt: string;
};

export type BudgetTagRule = {
  id: string;
  memberId: string | null;
  operator: "contains" | "starts_with" | "equals";
  pattern: string;
  tags: string[];
  createdAt: string;
};

export type BudgetDocument = {
  id: string;
  title: string;
  ownerIds: string[];
  memberIds: string[];
  members: BudgetMember[];
  shareCode: string;
  createdAt: string;
  updatedAt: string;
  customCategories?: BudgetCustomCategory[];
  categoryRules?: BudgetCategoryRule[];
  customTags?: BudgetCustomTag[];
  tagRules?: BudgetTagRule[];
  goals?: BudgetGoal[];
};

export type BudgetMonth = {
  id: string;
  month: string;
  incomes: BudgetIncome[];
  fixeds: BudgetFixedExpense[];
  entries: BudgetLedgerEntry[];
  savingsTarget?: number;
  createdAt: string;
  updatedAt: string;
  initializedFrom?: string | null;
};

File: android/toodl-share/app/src/main/java/com/toodl/share/data/model/Budget.kt
package com.toodl.share.data.model

data class BudgetDocument(
    val id: String = "",
    val title: String = "",
    val ownerIds: List<String> = emptyList(),
    val memberIds: List<String> = emptyList(),
    val members: List<BudgetMember> = emptyList(),
    val shareCode: String = "",
    val createdAt: String = "",
    val updatedAt: String = ""
)

data class BudgetMember(
    val id: String = "",
    val email: String? = null,
    val name: String? = null
)

data class BudgetMonth(
    val id: String = "",
    val month: String = "",
    val incomes: List<BudgetIncome> = emptyList(),
    val fixeds: List<BudgetFixedExpense> = emptyList(),
    val entries: List<BudgetLedgerEntry> = emptyList(),
    val savingsTarget: Double? = null,
    val createdAt: String = "",
    val updatedAt: String = "",
    val initializedFrom: String? = null
)

data class BudgetIncome(
    val id: String = "",
    val source: String = "",
    val amount: Double = 0.0
)

data class BudgetFixedExpense(
    val id: String = "",
    val name: String = "",
    val amount: Double = 0.0,
    val enabled: Boolean = true,
    val dueDay: Int? = null
)

data class BudgetLedgerEntry(
    val id: String = "",
    val amount: Double = 0.0,
    val category: String = "",
    val merchant: String? = null,
    val date: String = "",
    val isOneTime: Boolean = false,
    val tags: List<String> = emptyList()
)
